### ISSUE TITLE
@set Directive makes duplicate calls

### ISSUE THREAD

**Issue by marlongichie** [RobinRadic/blade-extensions#68]
When I used the `@set` directive like so :
```
@set($some_variable, $model->someFunction())
```

The `$model->someFunction()` is being called twice.

If I replace the directive with standard PHP tags like so:
```
<?php $some_variable = $model->someFunction(); ?>
```
Everything works normal... the `$model->someFunction()` is called once as expected.

Anyone else come across this issue?

Please advise.

**Comment by RobinRadic** [comment]
Hey,

It's a silly mistake:

```php
@set($some_variable, $model->someFunction())
```
transforms into
```php
$some_variable = $model->someFunction(); $__data['some_variable'] = $model->someFunction(); 
```
should transform into
```php
$some_variable = $model->someFunction(); $__data['some_variable'] = $some_variable; 
```

**fix**
```php
class SetDirective extends AbstractDirective
{
    protected $pattern = '/(?<!\w)(\s*)@NAME\s*\(\s*\${0,1}[\'"\s]*(.*?)[\'"\s]*,\s*([\W\w^]*?)\)\s*$/m';

    // protected $replace = '$1<?php \$$2 = $3; $__data[\'$2\'] = $3; ?>';
    protected $replace = '$1<?php \$$2 = $3; $__data[\'$2\'] = $2; ?>';
}
```

I'll release a patch soon

**Comment by marlongichie** [comment]
@RobinRadic was this update included in `v7.0` ???

**Comment by RobinRadic** [comment]
No,

You'd have to use this fix until i release a new version.

I intend to have the next release include the changes for Laravel 5.5, as well as this fix. But i'm currently hold up by another project, so it might take another week or 2.
